src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,149 @@
+package org.zalando.nakadi.service.timeline;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.config.NakadiSettings;
+import org.zalando.nakadi.config.SecuritySettings;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.PartitionStatistics;
+import org.zalando.nakadi.domain.Storage;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.exceptions.ForbiddenAccessException;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.exceptions.NotFoundException;
+import org.zalando.nakadi.exceptions.TimelineException;
+import org.zalando.nakadi.exceptions.TopicRepositoryException;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.TopicRepositoryHolder;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.db.StorageDbRepository;
+import org.zalando.nakadi.repository.db.TimelineDbRepository;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.view.TimelineRequest;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TimelineService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimelineService.class);
+    private static final String DEFAULT_STORAGE = "default";
+
+    private final SecuritySettings securitySettings;
+    private final EventTypeCache eventTypeCache;
+    private final StorageDbRepository storageDbRepository;
+    private final TimelineSync timelineSync;
+    private final NakadiSettings nakadiSettings;
+    private final TimelineDbRepository timelineDbRepository;
+    private final TopicRepositoryHolder topicRepositoryHolder;
+
+    @Autowired
+    public TimelineService(final SecuritySettings securitySettings,
+                           final EventTypeCache eventTypeCache,
+                           final StorageDbRepository storageDbRepository,
+                           final TimelineSync timelineSync,
+                           final NakadiSettings nakadiSettings,
+                           final TimelineDbRepository timelineDbRepository,
+                           final TopicRepositoryHolder topicRepositoryHolder) {
+        this.securitySettings = securitySettings;
+        this.eventTypeCache = eventTypeCache;
+        this.storageDbRepository = storageDbRepository;
+        this.timelineSync = timelineSync;
+        this.nakadiSettings = nakadiSettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.topicRepositoryHolder = topicRepositoryHolder;
+    }
+
+    public void createTimeline(final TimelineRequest timelineRequest, final Client client)
```

Comment:
It should be fine, I do not see any problem with locking here, shared resources should be thread safe. In addition to that this is the administrative resource which is used only by our team, I think we can figure out somehow not to switch timeline for the same event type simultaneously :)